lib/chef/resource.rb

```diff
@@ -416,6 +420,10 @@ def only_if(command = nil, opts = {}, &block)
     # @param block [Proc] A ruby block to run. Ignored if a command is given.
     #
     def not_if(command = nil, opts = {}, &block)
+      if command.is_a?(Hash) && opts.empty?
+        opts = command
+        command = nil
+      end
```

Comment:
yeah that was going to be my other comment is that i would have expected to see something like `opts = opts.dup.delete(:desc)` here.